### ISSUE TITLE
non-falsy-string breaks Attribute testing

### ISSUE THREAD

**Issue by joostdebruijn** [nunomaduro/larastan#1366]
- Larastan Version: 2.2.0 (on phpstan v1.8.4)
- `--level` used: 5

### Description

Since phpstan v1.8.3, the following error occurs:
```
Parameter $get of static method Illuminate\Database\Eloquent\Casts\Attribute<mixed,mixed>::make() expects 
  (callable(mixed, mixed=): string)|null,
  Closure(mixed, mixed): non-falsy-string given.
```
The `non-falsy-string` is introduced in phpstan v1.8.3. I think this test should pass, but it doesn't.

### Laravel code where the issue was found

```php

    /**
     * Returns the full name of a person.
     *
     * @return \Illuminate\Database\Eloquent\Casts\Attribute<string, never>
     */
    protected function fullname(): Attribute
    {
        return Attribute::make(
            get: function ($value, $attributes): string {
                if (! is_null($attributes['prefix'])) {
                    return $attributes['firstname'].' '.$attributes['prefix'].' '.$attributes['lastname'];
                }

                return $attributes['firstname'].' '.$attributes['lastname'];
            },
        );
    }

```

**Comment by szepeviktor** [comment]
Hello Joost! 👋🏻
Thank you for your report.

**Comment by joostdebruijn** [comment]
@szepeviktor Thanks for you blazing fast reply! I reported over there. Would you prefer to close this one for now or just leave it open untill they've responded upstream?

**Comment by ondrejmirtes** [comment]
The problem isn't the non-falsy-string part, the problem is that the code expects `callable(mixed, mixed=)` (one required parameter, one optional parameter), but your Closure requires two parameters.

This is a problem because the code that accepts the callable might call it only with one argument but when you give it a callable with two required parameters, PHP will crash.

**Comment by mreduar** [comment]
So this is a phpstan problem? I am experiencing exactly this problem as well.
```php
protected function fullImageUrl(): Attribute
{
    return Attribute::make(
        get: fn ($value, $attributes) => global_asset($attributes['image_url']),
    );
}

**Comment by szepeviktor** [comment]
AFAIK the given callback may not get a second argument `$attributes`.
https://github.com/nunomaduro/larastan/blob/8fbe7423782dac778f2383f8cc895b0da0faf733/stubs/Attribute.stub#L30

**Comment by szepeviktor** [comment]
Theory says it should have a default value `fn ($value, $attributes = [])`
But do not rely on me! I'm just a janitor here.

**Comment by mreduar** [comment]
> Theory says it should have a default value `fn ($value, $attributes = [])` But do not rely on me! I'm just a janitor here.

According to the [Laravel documentation](https://laravel.com/docs/9.x/eloquent-mutators#building-value-objects-from-multiple-attributes
) the second parameter is the attributes.

**Comment by nsrosenqvist** [comment]
How was this ever solved? I'm struggling with this too and I can't for the life of me figure out what the correct syntax is to specify here:
```php
    /**
     * @return \Illuminate\Database\Eloquent\Casts\Attribute<string, string>
     */
    protected function number(): Attribute
    {
        return Attribute::make(
            get: fn (string $value): string => "{$this->team->handle}-$value",
        )->shouldCache();
    }
```
The message I get is: `Parameter $get of static method Illuminate\Database\Eloquent\Casts\Attribute<mixed,mixed>::make() expects (callable(mixed, mixed): string)|null, Closure(string): non-falsy-string given.`

**Comment by calebdw** [comment]
Your generic should be `Attribute<string, never>` since you don't have a `set` closure

**Comment by nsrosenqvist** [comment]
@calebdw Thanks for the pointer, however, I changed that but am still getting the same error: `Parameter $get of static method Illuminate\Database\Eloquent\Casts\Attribute<mixed,mixed>::make() expects (callable(mixed, mixed): string)|null, Closure(string): non-falsy-string given.` Feels like I've tried every single combination of typing to get this one to pass. Any other ideas? If I change the stub in larastan to require `Closure` instead of `callable` it works.
